python/bridgestan/model.py

```diff
@@ -32,22 +33,33 @@ class StanModel:
 
     def __init__(
         self,
-        model_lib: str,
-        model_data: Optional[str] = None,
+        model_lib: Union[str, PathLike],
+        data: Union[str, PathLike, None] = None,
         *,
         seed: int = 1234,
+        stanc_args: List[str] = [],
+        make_args: List[str] = [],
         capture_stan_prints: bool = True,
+        model_data: Optional[str] = None,
     ) -> None:
         """
         Construct a StanModel object for a compiled Stan model and data given
         constructor arguments.
 
-        :param model_lib: A system path to compiled shared object.
-        :param model_data: Either a JSON string literal, a
+        :param model_lib: A system path to compiled shared object or a ``.stan``
+            file to be compiled.
+        :param data: Either a JSON string literal, a
```

Comment:
Would it be helpful to have listed `:param model_data:` also, with a note about it being deprecated in favor of `data`?

Reply:
I tend to not list renamed arguments in docstrings, but I'm happy to do so if you'd rather